src/Shared/FileUtilities.cs

```diff
@@ -361,37 +361,54 @@ internal static string MaybeAdjustFilePath(string value, string baseDirectory =
         {
             // Don't bother with arrays or properties or network paths, or those that
             // have no slashes.
-            if (NativeMethodsShared.IsWindows || string.IsNullOrEmpty(value) ||
-                value.StartsWith("$(") || value.StartsWith("@(") || value.StartsWith("\\\\") ||
-                value.IndexOfAny(Slashes) == -1)
+            if (NativeMethodsShared.IsWindows || string.IsNullOrEmpty(value)
+                || value.StartsWith("$(", PathComparison) || value.StartsWith("@(", PathComparison)
```

Comment:
Should these be `PathComparison` (`OrdinalIgnoreCase`) or straight `Ordinal`? Seems like the latter could be faster for these but maybe for non-cased characters it doesn't matter?

Reply:
Just ran a benchmark with the same StartsWith calls, Ordinal is definitely faster.